### ISSUE TITLE
Add name to manifest and reject other attributes

### ISSUE THREAD

**Issue by ewels** [nextflow-io/nextflow#871]
It would be great if `name` (workflow name) ~~and `author`~~ could be added as valid variables in the `manifest` scope.

Additionally, if non-standard variable names in the scope could trigger an error that would be helpful.

Thanks!

Phil

**Comment by pditommaso** [comment]
The `author` should be already there!

**Comment by ewels** [comment]
You’re right! My apologies.

**Comment by sven1103** [comment]
> Additionally, if non-standard variable names in the scope could trigger an error that would be helpful.

Hm, if I add a variable that is not part of the controlled vocabulary of the `manifest`, Nextflow already throws an error. I.e. when I add `age = '27' ` to the manifest scope, and try to access it in the `main.nf` with `workflow.manifest.age`, I get:

```
ERROR ~ No such variable: age
```
@ewels could you explain me, what you meant, maybe I did not get the issue properly.

Sven

**Comment by ewels** [comment]
True, trying to access it throws an error. I more meant to throw an error if you try to just define it.

**Comment by sven1103** [comment]
My current code would produce something like this: (dont know, why the warning messages are repeated though)

```
WARN: 'age' is not a valid manifest field!
WARN: 'pipelineVersion' is not a valid manifest field!
Launching `../hlatyping-workflow/main.nf` [elated_allen] - revision: 61c759c958
WARN: 'age' is not a valid manifest field!
WARN: 'pipelineVersion' is not a valid manifest field!
WARN: The config file defines settings for an unknown process: fastqc
WARN: 'age' is not a valid manifest field!
WARN: 'pipelineVersion' is not a valid manifest field!
WARN: Access to undefined parameter `reads` -- Initialise it to a default value eg. `params.reads = some_value`

```

**Comment by sven1103** [comment]
yes, and i dont know why honestly

**Comment by pditommaso** [comment]
A patch is available in the latest snapshot `0.32.1-SNAPSHOT`.

**Comment by pditommaso** [comment]
Released with version `18.10.1`.
